source/enc_ctx.c

```diff
@@ -31,34 +31,39 @@ int aws_cryptosdk_enc_ctx_init(struct aws_allocator *alloc, struct aws_hash_tabl
 }
 
 int aws_cryptosdk_enc_ctx_size(size_t *size, const struct aws_hash_table *enc_ctx) {
-    size_t serialized_len = 2;  // First two bytes are the number of k-v pairs
-    size_t entry_count    = 0;
+    AWS_PRECONDITION(AWS_OBJECT_PTR_IS_WRITABLE(size));
+    AWS_PRECONDITION(aws_hash_table_is_valid(enc_ctx));
 
+    const size_t entry_count = aws_hash_table_get_entry_count(enc_ctx);
+    if (entry_count > UINT16_MAX) return aws_raise_error(AWS_CRYPTOSDK_ERR_LIMIT_EXCEEDED);
+    if (entry_count == 0) {
+        // Empty context.
+        *size = 0;
+        AWS_POSTCONDITION(aws_hash_table_is_valid(enc_ctx));
+        AWS_POSTCONDITION(*size <= UINT16_MAX);
```

Comment:
We just set  *size =0 in the prev statements, what is this purpose of this post condition?